setup.cfg

```diff
@@ -16,8 +16,8 @@ data_file = .tox/.coverage
 
 [testenv:mypy]
 deps =
-    mypy==1.4.*
-    types-requests==2.28.*
+    mypy
```

Comment:
The reason for removing the version pins is to prevent future compatibility issues with upcoming Python releases (e.g., 3.13 or 3.14), similar to the `aiohttp` issue we resolved in this PR. The goal is to continue using mypy for type checking over time, rather than being tied to version 1.4.x. Additionally, the current latest mypy version is 1.11.2, and maintaining strict version pins would increase long-term maintenance costs as new releases emerge.

I understand that pinning versions can prevent potential breakage caused by changes in dependencies, as you pointed out. However, this increases the cost of adapting to new environments, so it’s a trade-off. I believe making it easier to adapt to new environments is beneficial for your personal project, but I will follow your decision.

Thank you!

Reply:
No, all dependencies must be pinned 

https://12factor.net/dependencies

Reply:
Understood. In that case, would the following versions be acceptable?
- `mypy~=1.4`
- `types-requests~=2.28`
- `ruff~=0.6`
- `flake8~=7.1`
- `pylint~=3.3`
- `isort~=5.13`

All tests on GitHub Actions have passed with these versions. Let me know if this works for you.

Reply:
NOTE: the tests fail due to version issues with Pylint and Flake8. Pylint version 3.0 or higher and Flake8 version 6.1.0 or higher are required.

Reply:
Okay, please update PR